frontend/src/metabase/search/components/SearchResult.unit.spec.js

```diff
@@ -23,6 +24,47 @@ function collection({
   return collection;
 }
 
+describe("SearchResult > Tables", () => {
+  it("only tables with initial_sync_status='complete' are clickable", async () => {
+    const table = { id: 1, display_name: "Table Name" };
+    const database = { id: 1, name: "Database Name" };
+    fetchMock.get("path:/api/table/1", table);
+    fetchMock.get("path:/api/database/1", database);
+    const result = {
+      model: "table",
+      name: "Complete Table",
+      initial_sync_status: "complete",
+      table_id: 1,
+      database_id: 1,
+      getUrl: () => `/table/1`,
+      getIcon: () => ({ name: "table" }),
+    };
+
+    const onClick = jest.fn();
+    renderWithProviders(<SearchResult result={result} onClick={onClick} />);
+    const link = screen.getByText("Complete Table");
+    fireEvent.click(link);
+    expect(onClick).toHaveBeenCalled();
```

Comment:
I'd split those tests into 2 (positive and negative)

Reply:
Done [234d898](https://github.com/metabase/metabase/pull/31651/commits/234d8985482ff0e0f82f7953ad233dd2b50fcda5)